File: Src/Icm.ContextConsole.Tests/FrontControllerTest.vb
Imports NUnit.Framework
Imports Icm.ContextConsole
Imports Icm.Localization
Imports Moq

<TestFixture>
Public Class FrontContextTest

    <Test>
    Public Sub ConstructorTest()
        Assert.That(Sub()
                        Dim fctl As IContext = New TestFrontContext
                    End Sub,
                    Throws.Nothing)
    End Sub

    <TestCase("asdf", "executecommand_actionnotfound", False)>
    <TestCase("help", Nothing, False)>
    <TestCase("quit", Nothing, True)>
    Public Sub ExecuteCommandTest(input As String, errorKey As String, expectedMustQuit As Boolean)
        Dim tw = New System.IO.StringWriter()
        Dim twerr = New System.IO.StringWriter()
        Dim inter = BuildInteractor({input}, tw, twerr)
        Dim mustQuit As Boolean
        Assert.That(Sub()
                        Dim fctl As IApplication = StandardApplication.Create(Of TestFrontContext)(interactor:=inter, intLocRepo:=New DictionaryLocalizationRepository())
                        mustQuit = fctl.ExecuteCommand()
                    End Sub,
                    Throws.Nothing)

        Assert.That(mustQuit, [Is].EqualTo(expectedMustQuit))
        If errorKey IsNot Nothing Then
            Assert.That(twerr.ToString, [Is].StringContaining(errorKey))
        End If
    End Sub

End Class

File: Src/Icm.ContextConsole.Tests/FrontControllerTest.cs

using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using NUnit.Framework;
using Icm.ContextConsole;
using Icm.Localization;
using Moq;

[TestFixture()]
public class FrontContextTest
{

	[Test()]
	public void ConstructorTest()
	{
		Assert.That(() =>
		{
			IContext fctl = new TestFrontContext();
		}, Throws.Nothing);
	}

	[TestCase("asdf", "executecommand_actionnotfound", false)]
	[TestCase("help", null, false)]
	[TestCase("quit", null, true)]
	public void ExecuteCommandTest(string input, string errorKey, bool expectedMustQuit)
	{
		dynamic tw = new System.IO.StringWriter();
		dynamic twerr = new System.IO.StringWriter();
		dynamic inter = BuildInteractor({ input }, tw, twerr);
		bool mustQuit = false;
		Assert.That(() =>
		{
			IApplication fctl = StandardApplication.Create<TestFrontContext>(interactor: inter, intLocRepo: new DictionaryLocalizationRepository());
			mustQuit = fctl.ExecuteCommand();
		}, Throws.Nothing);

		Assert.That(mustQuit, Is.EqualTo(expectedMustQuit));
		if (errorKey != null) {
			Assert.That(twerr.ToString, Is.StringContaining(errorKey));
		}
	}

}

//=======================================================
//Service provided by Telerik (www.telerik.com)
//Conversion powered by NRefactory.
//Twitter: @telerik
//Facebook: facebook.com/telerik
//=======================================================
